ui/src/app/views/workspace-list/component.ts

```diff
@@ -66,18 +69,25 @@ export class WorkspaceListComponent implements OnInit, OnDestroy {
                 const response: ErrorResponse = ErrorHandlingService.convertAPIError(error);
                 this.errorText = (response.message) ? response.message : '';
               });
-      } else {
-        this.billingProjectInitialized = false;
-        this.billingProjectQuery = setTimeout(() => {
-          this.profileStorageService.reload();
-        }, 10000);
-      }
-    });
+          // This may execute synchronously, no guarantee this has been assigned above yet.
+          if (this.profileSubscription) {
+            this.profileSubscription.unsubscribe();
+          }
+        } else {
+          this.billingProjectInitialized = false;
+          this.billingProjectQuery = setTimeout(() => {
+            this.profileStorageService.reload();
+          }, 10000);
+        }
+      });
     this.profileStorageService.reload();
   }
 
   ngOnDestroy(): void {
-    clearTimeout(this.billingProjectQuery);
+    if (this.billingProjectQuery) {
+      clearTimeout(this.billingProjectQuery);
+    }
+    this.profileSubscription.unsubscribe();
```

Comment:
Should we add an if check here?

Reply:
I think we're OK without here. `profileSubscription` always gets set in `ngOnInit`, which I think is safe to assume was called before `ngOnDestroy`. Also `unsubscribe()` is idempotent (we may have already unsubscribed above).

Reply:
Yeah, I wasn't sure on the idempotence. Sounds good!